lib/assets/csv_script.rb

```diff
@@ -0,0 +1,64 @@
+require 'roo'
+require 'csv'
+
+XLSX_PATH = ARGV[0] || "lib/assets/initial-list.xlsx"
+CSV_PATH = ARGV[1] || "lib/assets/pre-chunk-content.csv"
+NEW_CSV_ROOT = ARGV[2] || "lib/assets/2023-01-12-pilot-SNAP-warning-notice_"
+
+xlsx = Roo::Excelx.new(XLSX_PATH)
+
+# Write the data to a new CSV file (no BOM by default)
+xlsx.to_csv(filename=CSV_PATH)
+
+# Optional: explicitly remove BOM if opening a csv that was saved from excel
+# spreadsheet = Roo::Spreadsheet.open("my-csv.csv", { csv_options: { encoding: 'bom|utf-8' } })
+
+header_renames = {
+  "EDG_NUM" => "case_id",
+  "HH Client ID" => "client_id",
+  "HH First Name" => "first_name",
+  "HH Phone Number" => "phone_number",
+  "HH Primary Langauage" => "primary_language",
+}
+csv_data = CSV.read(
+  CSV_PATH,
+  :headers => true,
+  :header_converters => lambda { |header| header_renames[header] || header } # Rename columns
+)
+# Add column preferred_language using column 6 ("HH Primary Langauage")
+csv_data.each do |row|
+  row["preferred_language"] = row[6] == "SPA" ? "es" : "en"
+end
+
+# Add column submit_date
+csv_data.each do |row|
+  row["submit_date"] = row[6] == "SPA" ? "20 de enero" : "January 20"
+end
+
+# Remove columns
+csv_data.delete("CASE_NUM")
+csv_data.delete("HH Last Name")
+csv_data.delete("HH DOB")
+csv_data.delete("TFA")
+csv_data.delete("LTSS")
+
+# Write non-shuffled, pre-chunked data to a place holder csv
+CSV.open(CSV_PATH, "w", :write_headers=> true) do |csv|
```

Comment:
Do we need the intermediate file?

Reply:
I use an intermediate file because at this point the script switches from processing the data as a csv table to processing it as lines in a file. It is a subtle difference, but makes it much easier to preserve the headers, shuffle, and chuck the file lines as an array of strings rather than the CSV::Table object from roo.